modules/llrt_http/src/fetch.rs

```diff
@@ -202,10 +203,14 @@ fn build_request(
         req = req.header("accept", "*/*");
     }
 
-    req.body(BoxBody::new(
-        body.map(|b| b.body.clone()).unwrap_or_default(),
-    ))
-    .or_throw(ctx)
+    let body_unwrap = body.map(|b| b.body.clone()).unwrap_or_default();
+
+    if !detected_headers.contains("content-length") {
+        let body_len = body_unwrap.size_hint().exact().unwrap_or_default();
+        req = req.header("content-length", body_len.to_string());
+    }
+
+    req.body(BoxBody::new(body_unwrap)).or_throw(ctx)
```

Comment:
Nice finding! Should be set it for all request tho? GET, DELETE are maybe questionable?

Reply:
looks like should be set only if body is not empty,
at least I noticed such behaviour from postman, it auto sets content length if body is not empty, regardless of method

Reply:
I am 95% sure you dont need to set a content length explicitly. This is a bad practice anyway since the logic for content length is actually complex when you have to take into account compression and streaming (TE chunked).
This is handle by hyper in this function: https://github.com/hyperium/hyper/blob/0bd4adfed2bf40dda6867260ded05542d5b921c5/src/proto/h1/role.rs#L1484
But this is only called if the body has a known length, otherwise it will default to TE chunked:
https://github.com/hyperium/hyper/blob/0bd4adfed2bf40dda6867260ded05542d5b921c5/src/proto/h1/role.rs#L1298
It is possible that something broke when we previously switched to a BoxBody, but the solution is not to set an explicit header, we should find the root cause of the issue.

Reply:
The only way for a missing CL is if the body doesnt correctly implement the size hint which you can see the logic here:
https://github.com/hyperium/hyper/blob/0bd4adfed2bf40dda6867260ded05542d5b921c5/src/proto/h1/dispatch.rs#L339-L351

Note that it is a different logic for http/2 than http/1.1, I didnt dig into it but I would be very very very surprised if they didn't handle the CL.